adage/adage/settings.py

```diff
@@ -0,0 +1,147 @@
+"""
+Django settings for adage project.
+
+Generated by 'django-admin startproject' using Django 2.2.5.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/2.2/ref/settings/
+"""
+
+import os
+import yaml
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# Read secrets from YAML file
+path = os.path.join(BASE_DIR, 'adage', 'config.yml')
+with open(path) as read_file:
+    config = yaml.full_load(read_file)
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = config.get('secret_key', 'django_secret_key')
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = config.get('debug', True)
+
+ALLOWED_HOSTS = config.get('allowed_hosts', [])
+
+# Application definition
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'rest_framework',
+    'django_filters',
+    'corsheaders',
+    'organisms',
+    'genes',
+    'analyze',
```

Comment:
Agree. I simply copied this line from original Adage settings. `analysis` is one option, but it seems a little too generic. @cgreene, do you have any preference on the name of this Django app?

Reply:
Is this app essentially the place where most of the endpoints live? It's too generic for me to remember what goes in it now. 😭

Reply:
@cgreene: Yes, you are right.

Reply:
In a new django project, would this just be called `adage`? I think the django devs might have changed how apps were named by default a bit after this project was created.

Reply:
In Django's jargon, `adage` is the project's name, `analyze`, `genes`, `organisms` are App's name. The app's name doesn't really matter to the frontend. It only matters to backend development. So we can name it anything, but still we want it to be a reasonable one.

Reply:
I'm really glad we're doing this review. This is the first time for this particular naming decision because the first time around actually predates the GreeneLab code review policy!

I jabbed the original developer in the ribs and asked "what were you thinking?" (We go [way back](https://bitbucket.org/greenelab/adage-server/commits/5e828ba006773fc0212493ddc365b14072d08216).) He mumbled something about it seeming good at the time because we didn't really know what the data models were going to look like.

The naming is indeed generic because it encompasses everything. I wonder if it makes sense to split the backend into two Django "apps" because there *is* something of a logical split between the annotated "source data" (`Samples`, essentially, which we pulled from [ArrayExpress](https://www.ebi.ac.uk/arrayexpress/) queries, along with `Experiment`, and all of the `SampleAnnotation` models) and "ADAGE model data" (`MLModel`, `Signature`, `Activity`, and the rest). That separation of concerns would help our system design by giving a stronger core purpose to the two parts of the backend.

We should also keep in mind that the Django framework was not originally written for single page apps. Our goal here should be to make the best use of the capabilities it provides without getting too hung up on parts that are not relevant to us as we aim to build a solid REST API.

Maybe we are wandering outside the scope of this pull request. Does it make sense to resolve this question via an issue?

Reply:
Update: I suggest we follow the default Django 2.2 project layout @cgreene [points out](https://docs.djangoproject.com/en/2.2/intro/tutorial01/#creating-a-project) and dispense with the idea of a separate "app" (currently named `analyze`). The possibility of separate app modules that Django offers are very useful as reusable modules and they make sense for something portable like `genes` and `organisms`, but the bulk of the Adage API is probably inseparable from the rest of the back end code.